File: src/SupremeCourtOpinionMetadata.py
from Metadata import Metadata

class SupremeCourtOpinionMetadata(Metadata):
    '''
    SupremeCourtOpinionMetadata Class for Word Cloud Project
    
    Daniel Klein
    Computer-Based Honors Program
    The University of Alabama
    10.16.2013
    
    This specializes the generic Metadata class to apply to Supreme Court 
    opinions. 
    '''


    def __init__(self):
        '''
        This overrides the original dummy constructor in the Metadata class.
        '''
        self.field_names = (["case_title", "case_num", "case_lexis_cite",
                             "case_full_cite", "case_dates", "case_disposition",
                             "opinion_author"])
        self.case_title = ""
        self.case_num = ""
        self.case_us_cite = ""
        self.case_supreme_court_cite = ""
        self.case_lawyers_ed_cite = ""
        self.case_lexis_cite = ""
        self.case_full_cite = ""
        self.case_dates = []
        self.case_disposition= ""
        self.opinion_author = ""
        
        
    
        

File: src/core/java/SupremeCourtOpinionMetadata.java
/**
 * 
 */
package core.java;

import static java.util.Arrays.asList;

/**
 *   SupremeCourtOpinionMetadata Class for Word Cloud Project (Java)
 *   
 *   Daniel Klein
 *   Computer-Based Honors Program
 *   The University of Alabama
 *   5.10.2014
 *   
 *   This specializes the generic Metadata class to apply to Supreme Court 
 *   opinions. 
 */
public class SupremeCourtOpinionMetadata extends Metadata {
	
	
	/**
	 * Constructor -- initializes every metadata field to empty string
	 */
	public SupremeCourtOpinionMetadata()
	{
		this.fieldNames = asList("Case Title", "Case Number", "US Citation",
		                         "Supreme Court Citation", "Case Lawyers Ed Citation",
		                         "Lexis Citation", "Full Citation", "Case Dates", 
		                         "Case Disposition", "Opinion Author", 
		                         "Opinion Type");
		for (String fieldName: this.fieldNames)
		{
			this.fields.put(fieldName, "");
		}
	}
	
	
	public String toString()
	{
		String returnString = "";
		for (String fieldName: this.fieldNames)
		{
			returnString += fieldName + " : " + this.fields.get(fieldName) + "\n";
		}
		return returnString;
	}

	
	public void print()
	{
		System.out.println(this.toString());
	}
	
	
	public void printFields()
	{
		String printString = "";
		for (String fieldName: this.fieldNames)
		{
			printString += fieldName + "\n";
		}
		System.out.println(printString);
	}
}
